androidtv/constants.py

```diff
@@ -15,6 +15,8 @@
 INTENT_LAUNCH = "android.intent.category.LAUNCHER"
 INTENT_HOME = "android.intent.category.HOME"
 
+# Specify application identifier as CURRENT_APP from dumpsys output previously assigned to CURRENT_APP
+VAR_CURRENT_APP_EXTRACTION = 'CURRENT_APP=${CURRENT_APP#*ActivityRecord{* * } CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && CURRENT_APP=${CURRENT_APP%\\}*}'
```

Comment:
This isn't used.

Reply:
Actually, I guess it is used. Chrome on my phone only found it once on this page. So maybe it's OK.

Reply:
Actually this is the core "logic" in regard to extracting the application identifier from the dumpsys output, either it being from `dumpsys window windows` or `dumpsys activity a .`.

It does the following operations:
1. `${CURRENT_APP#*ActivityRecord{* * }`
Attempt to extract application activity from `mResumedActivity` or `mFocusedApp`
2. `${CURRENT_APP#*{* * }`
Fallback to extract application activity from `mCurrentFocus`, in case `mFocusedApp` is not defined
3. `${CURRENT_APP%%/*}`
Remove extraneous activity details for extracted application, leaving only identifier
4. `${CURRENT_APP%\\}*}`
Delete trailing curly brace from identifier of extracted application, in case application has no activity